.github/workflows/build_and_run_test_app.yml

```diff
@@ -0,0 +1,111 @@
+name: Build and Run Test Application
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build Test App
+    strategy:
+      matrix:
+        idf_ver: ["release-v4.3", "release-v4.4", "latest"]
+        idf_target: ["esp32", "esp32c3", "esp32s3"] # @todo ESP32S2 has less RAM and the test_app will not fit
```

Comment:
One note about this — I think it's not very convenient to specify the target list for test apps here in CI configuration. It will probably get quite messy if the repository gets more test applications, and each test application will have its own set of targets.

We could consider using find_apps.py and build_apps.py scripts from IDF to perform builds for all targets, but in that case we would need to support the `merge_bin` step as part of build_apps.py.

(No action needed for now, just bringing this up. I'm afraid the missing tests for S2 will be easy to forget about.)